### ISSUE TITLE
1.16.1 Can't open Potion Bag GUI on Server

### ISSUE THREAD

**Issue by kingofrunes** [zabi94/ExtraAlchemy#87]
Having a weird issue where when I shift right click the Potion Bag I can't open it on the server. It works on Single Player.

Not sure if this is related, but after I tried to open it and logged out and back into the server, I ran into this error that required a server reboot: "Internal Exception: io.netty.handler.code.DecoderException: Badly compressed packet - size of 2308003 is larger than protocol maxium of 2097152". Happened twice, each time after I attempt to open the potion bag several times. Haven't seen the error come up since.

Mods: 
Extra Alchemy 1.2.1
Fabric 0.16.2? Hard to tell. Fabric Loader is 0.9.1+build.205. The log spits out fabric@0.16.2+build.385-1.16.1

**Comment by kingofrunes** [comment]
Update. Had the server admin try and replicate and these are the steps they did.
1. Attempt to open bag, nothing
2. Log out, log back in
3. Attempt to open bag, worked no issue.
4. Log out, log back in (to test the bad packet error. No error.

I replicated the same process as admin and was able to replicate his behavior. Really weird. Maybe it's when I was attempting repeatedly to open it when it wasn't working is what was causing the bad packet error

**Comment by zabi94** [comment]
Can you include a link to the pastebin pages with the server and client log files? I need to check if something useful pops up there.
I also need the server and client mod lists

**Comment by zabi94** [comment]
Also, please next time follow the bug report template, it includes everything I need to know

**Comment by kingofrunes** [comment]
[Info: 2020-08-31 12:12:56.7915676: GameCallbacks.cpp(163)] Game/net.minecraft.class_3929 (main) Warn Trying to open invalid screen with name: Potion Bag

That's what I see client side.

This is the client side mods
Loading 162 mods: fermion@2.1.191, fabric-keybindings-v0@0.2.0+3fa9f7c5b0, notenoughcrashes@2.1.0+1.16.1, cloth-common-events-v1@1.0.8, fabricloader@0.9.1+build.205, leaf-decay@1.2.0-1.16.1, artifice@0.8.1+1.16.1, sleepvote@${version}, cloth-durability-bar-api-v1@1.0.8, iron-jetpacks@0.2.5, fabric-dimensions-v1@1.0.0+a71b305302, libgamerule@${version}, soulbound@1.0.1, extrapieces@2.9.0, fabric-game-rule-api-v1@1.0.0+fe81e12502, fermion-modkeys@1.8.180-snapshot, staffofbuilding@1.2.0-1.16, glassential@1.16.1-1.1.4, fabric-api-base@0.1.3+12a8474cb0, reborncore@4.3.4+build.64, controlling@1.1.2, xaeroworldmap@1.9.0, fiber@0.23.0-2, creativebuttonmover@0.2.5, roughlyenoughresources@2.0.2, ninjaphenix-container-lib@1.1.5, cardinal-components-level@2.4.2, astromine@1.5.2, org_jetbrains_kotlin_kotlin-stdlib@1.3.71, waila@1.9.22, fabric-blockrenderlayer-v1@1.1.4+c6a8ea8902, org_jetbrains_kotlin_kotlin-stdlib-jdk7@1.3.71, libblockattributes_items@0.7.1-pre.4, fabric@0.16.2+build.385-1.16.1, respawnablepets@4, fabric-networking-v0@0.1.10+e00ecb5f02, org_codehaus_groovy_groovy@3.0.3, wrenchable@1.0.0+1.16.1, techreborn@3.4.5+build.88, cotton-config@1.0.0-rc.7, fabric-screen-handler-api-v1@1.1.0+97324d1102, modmenu@1.14.5+build.30, dataloader@2.1.0+mc1.16.1, columns@1.1.0, simpledrawers@1.8.0-1.16.1, phosphor@0.6.0+build.7, cloth-utils-v1@1.0.8, fabric-tool-attribute-api-v1@1.2.0+55e0cc5f02, org_jetbrains_kotlin_kotlin-stdlib-jdk8@1.3.71, enderlift@1.0.5, cloth-client-events-v0@1.0.8, blockus@1.6.2, fabric-biomes-v1@0.2.7+059ea86602, fabric-events-interaction-v0@0.3.3+7066030fb0, fabric-crash-report-info-v1@0.1.2+b7f9825d02, whereisit@1.8.1, loadcatcher@2.0.0, curios@1.16.1-0.5, fabric-language-kotlin@1.3.71+build.1, patchouli@1.16-39-FABRIC, mousewheelie@1.5.0+mc1.16.1, fabric-key-binding-api-v1@1.0.1+f404f3be02, cloth-config2@4.6.0, org_jetbrains_kotlinx_kotlinx-coroutines-jdk8@1.3.5, fabric-resource-loader-v0@0.2.6+f41e209802, chainmail@0.3.0, org_jetbrains_kotlin_kotlin-reflect@1.3.71, expandedstorage@5.3.16, universalcomponents@0.7.2+1.16-pre2, cardinal-components-entity@2.4.2, magna@1.3.2-1.16.1, libblockattributes_fluids@0.7.1-pre.4, staticdata@1.1.2, fabric-commands-v0@0.2.0+52d3083602, fabric-registry-sync-v0@0.3.8+7dba2d6c02, adorn@1.10.0+1.16.1, cardinal-components-base@2.4.2, lightoverlay@5.4.1, fabric-mining-levels-v0@0.1.2+b764ce9902, fabric-item-groups-v0@0.2.0+438f963602, fabric-models-v0@0.1.0+dfdb52d602, extraalchemy@1.2.1, shulkerboxtooltip@2.1.0+1.16.1, org_jetbrains_annotations@17.0.0, fabric-renderer-api-v1@0.2.13+eae12eb802, parchment@1.0.2+1.15.2, linkedstorage@1.3.5-1.16, terrestria@2.0.4+build.50, libcd@2.4.1+1.16-pre2, cardinal-components-block@2.4.2, org_jetbrains_kotlinx_kotlinx-coroutines-core@1.3.5, simpleteleporters@2.2.0, lithium@0.5.1, fabric-containers-v0@0.1.8+045df74f02, fabric-object-builder-api-v1@1.5.6+a2d21ddd02, leafmealone@2.1, fabric-rendering-data-attachment-v1@0.1.3+b7f9825d02, mm@2.0, campanion@1.1.13+build.35, curioofundying@1.16.1-1.0, cardinal-components-item@2.4.2, fabric-textures-v0@1.0.4+eae12eb802, wolveswitharmor@1.5.1-1.16.1, fabric-rendering-fluids-v1@0.1.8+b7084faa02, reach-entity-attributes@1.0.1, cloth-scissors-api-v1@1.0.8, cotton-logging@1.0.0-rc.4, fabric-lifecycle-events-v1@1.2.0+e83e061c02, cardinal-components-chunk@2.4.2, cardinal-components-util@2.4.2, fabric-renderer-registries-v1@2.0.1+5a0f9a6002, tinyfuel@0.3.0, quickshulker@1.1.5-1.16, fabric-loot-tables-v1@1.0.0+386eb69e02, cloth-armor-api-v1@1.0.8, fabric-particles-v1@0.2.1+0a6f2a7002, artofalchemy@1.0.0-rc2+1.16.1, libblockattributes_core@0.7.1-pre.4, terraform@2.1.7+build.31, ynet@0.2.0, slotlink@1.0.3+minecraft-1.16.1, okzoomer@3.0.1-20w12a, minecraft@1.16.1, fabric-networking-blockentity-v0@0.2.5+b50ffc7b02, boringbackgrounds@2.1.0-1.16.pre3, cloth-basic-math@0.5.1, jankson@3.0.0+j1.2.0, fabric-command-api-v1@1.0.8+5ce5339802, fatxporbs@0.0.7, static-content@1.0.0-1.16.1, amecsapi@1.0.2+1.16-rc1, fabric-renderer-indigo@0.3.4+34d6c87102, spinnery@3.1.14, roughlyenoughitems@4.10.5, fabric-item-api-v1@1.0.0+16acbe5b02, shapes@2.0.0+build.10, kyrptconfig@1.0.4-1.16, libgui@2.0.1+1.16.1, slight-gui-modifications@1.2.0, xaerominimap@20.19.0, fabric-rendering-v0@1.1.0+5341049002, fabric-rendering-v1@1.1.2+346247d702, autoconfig1u@3.2.0-unstable, harvest@1.2.12, iknowwhatimdoing@3.0.0, fabric-content-registries-v0@0.1.9+059ea86602, cardinal-components-world@2.4.2, fabric-tag-extensions-v0@0.2.7+a4c57d8e02, blue_endless_jankson@+, inmis@1.2.0-1.16.1, gravestones@1.4.0, window_title_changer@2.0.0-1.16.1, appleskin@1.0.11, cloth-api@1.0.8, ring_of_attraction@1.0.21, cinderscapes@1.0.7+build.43-1.16.1, fabric-events-lifecycle-v0@0.2.0+16acbe5b02, libstructure@1.3, org_codehaus_groovy_groovy-jsr223@3.0.3, team_reborn_energy@0.1.0, fabric-object-builders-v0@0.6.1+a2d21ddd02, tweed@3.0.0-beta.9

I asked the server admin for the info from the server, see if there's anything there.

Mod list from server.
Adorn-1.10.0+1.16.1.jar
Controlling-For-Fabric-1.1.2.jar
Harvest-fabric-1.16.1-1.2.12-26.jar
Hwyla-fabric-1.16.1-1.9.22-75.jar
Patchouli-1.16-39-FABRIC.jar
RebornCore-1.16-4.3.4+build.64.jar
RingOfAttraction-Fabric_1.16.1-1.0.21.jar
RoughlyEnoughItems-4.10.5.jar
Soulbound-fabric-1.16.1-1.1.1-7.jar
TechReborn-1.16-3.4.5+build.88.jar
Towelette-4.1.5+1.16.1.jar
WhereIsIt-1.16.1-1.8.1.jar
appleskin-mc1.16-fabric-1.0.11.jar
artofalchemy-1.0.0-rc2+1.16.1.jar
astromine-1.5.2+fabric-1.16.1.jar
blockus-1.6.2.jar
campanion-1.1.13+build.35.jar
cinderscapes-1.0.7+build.43-1.16.1.jar
cloth-api-1.0.8.jar
columns-1.1.0.jar
config-2-4.6.0.jar
curioofundying-fabric-1.16.1-1.0.jar
curios-fabric-1.16.1-0.5.jar
data-loader-2.1.0+mc1.16.1.jar
enderlift-1.0.7.jar
expandedstorage-5.3.16+1.16.jar
extraalchemy-fabric-1.16.1-1.2.1+build16.jar
extrapieces-2.9.0+1.16.1.jar
fabric-api-0.16.2+build.385-1.16.1.jar
fabric-language-kotlin-1.3.71+build.1.jar
fatxporbs-0.0.7+1.16-rc1.jar
glassential-fabric-1.16.1-1.1.4.jar
gravestones-1.4.0.jar
inmis-1.2.0-1.16.1.jar
iron-jetpacks-0.2.5.jar
leaf-decay-1.2.0-1.16.1.jar
leafmealone-2.1.jar
linkedstorage-1.3.5-1.16.jar
lithium-fabric-mc1.16.1-0.5.1.jar
notenoughcrashes-2.1.3+1.16.1.jar
phosphor-fabric-mc1.16.1-0.6.0+build.7.jar
quickshulker-1.1.5-1.16.jar
rer-2.0.2.jar
respawnablepets-4.jar
shulkerboxtooltip-2.1.0+1.16.1.jar
simpledrawers-1.8.0-1.16.1.jar
simpleteleporters-2.2.0.jar
sleepvote-1.8+1.16.1.jar
slotlink-1.0.3+minecraft-1.16.1.jar
staffofbuilding-1.2.0-1.16.jar
terrestria-2.0.4+build.50.jar
tinyfuel-fabric-0.3.0.jar
wolves-with-armor-1.5.1-1.16.1.jar
ynet-0.2.0+1.16.1.jar

**Comment by kingofrunes** [comment]
Nothing useful server side from the logs. Only thing is the client which is most likely the source of the issue. The `[Info: 2020-08-31 12:12:56.7915676: GameCallbacks.cpp(163)] Game/net.minecraft.class_3929 (main) Warn Trying to open invalid screen with name: Potion Bag` message.

**Comment by rikka0w0** [comment]
`Trying to open invalid screen with name: Potion Bag` is printed by `net.minecraft.client.gui.screen.Screens.open(@Nullable ContainerType<T> type, MinecraftClient client, int id, Text name)` because the ContainerType variable is null. Its value comes from `OpenContainerS2CPacket`:
```java
   public ContainerType<?> getContainerType() {
      return (ContainerType)Registry.CONTAINER.get(this.containerId);
   }
```
Looks like the container is not registered properly on the client side. But I cannot spot any problem in the usage of Fabric Gui API in this mod.

**Comment by rikka0w0** [comment]
The fix works. In 1.16.1, the problem can be fixed by adding a line: 
```java
zabi.minecraft.extraalchemy.screen.ModScreenHandlerTypes.POTION_BAG.toString();
```
at the end of `zabi.minecraft.extraalchemy.ExtraAlchemy.onInitialize()`.

I'm 99% sure this is a Fabric API bug. I will start an issue on their repo.

Actually this might not be a bug. The problem is caused by the way we use the API.

**Comment by zabi94** [comment]
No, it is not an api bug. It was extra alchemy that had the registration happen at class init instead of mod init. By adding that line you load the class at the right moment instead of at world join.
That confirms the fix I pushed yesterday

**Comment by zabi94** [comment]
Anyway, it will be fixed in the next release for 1.16.2

**Comment by zabi94** [comment]
Fix available on curseforge
